modules/lang-painless/src/test/java/org/elasticsearch/painless/WhenThingsGoWrongTests.java

```diff
@@ -840,4 +840,8 @@ public void testInstanceMethodNotFound() {
         iae = expectScriptThrows(IllegalArgumentException.class, () -> exec("doesNotExist(1, 'string', false)"));
         assertEquals(iae.getMessage(), "Unknown call [doesNotExist] with [3] arguments.");
     }
+
+    public void testArrayToArrayException() {
+        expectScriptThrows(IllegalArgumentException.class, () -> exec("return new String[] {'a'}.method()"));
```

Comment:
Would you please add a snippet to compare the exception message? We throw `IllegalArgumentException` for a number of different errors, so I would like to make sure this is throwing the error we expect.

Reply:
Done